lib/Settings.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
+var nconf = require('nconf');
+var Promise = require('bluebird');
+
+var defined = Cesium.defined;
+
+nconf.env('__')
+    .file({
+        file: 'config.json'
+    })
+    .defaults({
+        port: 5000,
+        listenPath: '/'
+    })
+    .required([
+        'secret'
+    ]);
+
+var Settings = {};
+Settings.loadRepositoriesSettings = function() {
+    var name;
+    return fsExtra.readJson('repositories.json')
+    .then(function(repositoryJson) {
+        if (!(repositoryJson instanceof Array)) {
+            return Promise.reject('Must put at least one repository in `repositories.json` as Array');
+        }
+        Settings._repositories = [];
+        repositoryJson.forEach(function(repository) {
+            name = repository.name;
+            if (!defined(name)) {
+                return Promise.reject('Repository must have `name` field!');
+            }
+            if (!defined(repository.gitHubToken)) {
+                return Promise.reject('Repository ' + name + ' must have a `gitHubToken`');
+            }
+            Settings._repositories.push(name);
+            Settings.set(name, 'gitHubToken', repository.gitHubToken);
+            Settings.set(name, 'remindForum', repository.remindForum);
+        });
+    });
+};
+
+Settings.set = function(repositoryName, key, value) {
+    nconf.set(['repositories', repositoryName, key].join(':'), value);
```

Comment:
Feel free to simplify this code, there's no reason to store the repository information in nconf since it's all abstracted away  in the `Settings` object now.  You can actually just exposed the parsed `repositories.json` object as a property of the `Settings` object and be done with it (unless I'm missing something).